components/bl-gauge-chart-component/src/index.js

```diff
@@ -1,13 +1,29 @@
+import { useState, useEffect } from 'react';
+
 import { useChartData } from './use-chart-data';
 
 const { cn } = BackendlessUI.CSSUtils;
+const GOAL = 'GOAL';
 
 export default function GaugeChartComponent({ component }) {
   const { classList, display, disable, goal, progress, style } = component;
 
-  const { shownGoal, shownProgress, angleFillStyle, progressPercentage } = useChartData(goal, progress);
+  const [validateGoal, setValidateGoal] = useState(1);
+  const [validateProgress, setValidateProgress] = useState(0);
+
+  component.getGoal = () => validateGoal;
+  component.setGoal = goal => setValidateGoal(validate(goal, 'GOAL'));
+  component.getProgress = () => validateProgress;
+  component.setProgress = progress => setValidateProgress(validate(progress));
+
+  useEffect(() => {
+    setValidateGoal(validate(goal, 'GOAL'));
+    setValidateProgress(validate(progress));
+  }, [goal, progress]);
```

Comment:
Could we move all validations to useChartData? That could helps to avoid some useState-s.